### ISSUE TITLE
Unable to login to github 

### ISSUE THREAD

**Issue by fox-john** [github/vscode-github-actions#77]
**Describe the bug**
I'm unable to log-in to github through Vscode. I don't have error message. The login step in my browser is on success, but redirect to vs-code don'"t log me. 

I have try the second way this bottom notification, but same effect. 

![image](https://user-images.githubusercontent.com/1273438/229118700-5178e086-131b-450f-9f0a-38e933f4a290.png)
![image](https://user-images.githubusercontent.com/1273438/229118713-bf1ba3ad-4e84-4034-92a4-8ed290f0d2d5.png)

**To Reproduce**
- Install WSL
- Login trough WSL

**Expected behavior**
Login to github and show all workflow

**Screenshots**
![image](https://user-images.githubusercontent.com/1273438/229118932-051a0d7d-b396-42b0-8b8e-32a6f9c3a3c8.png)

**Extension Version**
`v0.25.3`

**Additional context**

**Comment by OpenCode** [comment]
Here too.
Ubuntu 22.04.2 LTS

**Comment by grzegorzjudas** [comment]
Same issue, MacBook Pro M1, Ventura 13.0, vscode 1.76.0. I get redirected to Github, proceed with login normally and when I confirm opening the vscode link, vscode still shows "Sign in to GitHub" with a notification saying that if I have trouble, I can try different method. Clicking yes opens `http://127.0.0.1:53220/` that only says "You are signed in" and that's it, nothing changes.

**Comment by grzegorzjudas** [comment]
I somehow fixed it - I opened another project/repo in different window and instead of a "Sign in to Github" button I got "Unable to connect to Github API. Do you have internet connection?". I then switched to the previous project, reloaded window and it loaded fine.

**Comment by inomdzhon** [comment]
Same problem((

MacOS 13.3 (22E252)

Version: 1.77.1 (Universal)
Commit: b7886d7461186a5eac768481578c1d7ca80e2d21
Date: 2023-04-04T23:23:33.779Z (6 days ago)
Electron: 19.1.11
Chromium: 102.0.5005.196
Node.js: 16.14.2
V8: 10.2.154.26-electron.0
OS: Darwin arm64 22.4.0
Sandboxed: No

**Comment by KetchupOnMyKetchup** [comment]
@fox-john - I just tried this in WSL Ubuntu 22.04 and it worked for me, here is a demo of that specific environment:
![vscodewsldemo](https://github.com/github/vscode-github-actions/assets/7976517/503973b8-016a-4756-85f0-0e4ab4ecbbcf)

We have also tested this on Windows, Mac Intel, and Mac M1 and it worked.

@fox-john @inomdzhon @OpenCode @MetRonnie -

Would you please try:
1. Uninstall `v0.25.3` and install the new version `v0.25.4` and login
2. Try reloading VS Code
  ![image](https://github.com/github/vscode-github-actions/assets/7976517/5606a15a-482e-4192-8822-8d2ef96c42bd)
3. Try disabling any other extensions to see if they are interfering. Then reload VS Code.
4. Do you have any Wiki repos in your GitHub repo? If so these could be the issue and we should make a bug for this. 
5. Lastly, could you check the Output tab in VS Code and select GitHub Actions in the dropdown?  Could you send us a copy of any errors there?
  ![image](https://github.com/github/vscode-github-actions/assets/7976517/e1ca53ac-cbb4-4be2-985c-f32698081185)

Please let me know if any of these steps work, if so which one. And if not, what error messages you have so I can debug. Thank you so much for your help!

**Comment by MetRonnie** [comment]
I am able to login with v0.25.4 👍

**Comment by KetchupOnMyKetchup** [comment]
> I'm using `0.25.4`. When I try to sign in my browser is opened to `https://github.com/login/oauth/authorize?client_id=xxx` and the extension output shows HTML indicating I've hit a rate limit
> 
> ```
> 2023-04-12 08:53:02.396 [info] Logging in for the following scopes: repo workflow
> 2023-04-12 08:53:02.397 [info] Trying without local server... (repo workflow)
> 2023-04-12 08:53:03.582 [info] Exchanging code for token...
> 2023-04-12 08:53:03.937 [error] GitHubTokenExchangeError: <!DOCTYPE html>
> ...
> <p>You have exceeded a secondary rate limit.<br><br>
> 
> at t.GitHubServer.u (c:\Users\...\github-authentication\dist\extension.js:2:1018998)
> 	at process.processTicksAndRejections (node:internal/process/task_queues:96:5)
> 	at async c:\Users\...\github-authentication\dist\extension.js:2:1013850
> ```

@rlaveycal Could you please try this again and see if it is a transient issue or not and let me know? Thanks!

**Comment by rlaveycal** [comment]
Yes it's transient. I have logged in successfully. It's annoying however, and it would be nice if the error was properly surfaced rather than having to trawl the logs.

**Comment by KetchupOnMyKetchup** [comment]
@inomdzhon - I'm sorry to hear that it isn't working for you still. 

Could you please retry the steps with a different GitHub repo and see if that works? I wonder if there is something specific about the repo you are trying to open that is causing this issue? If that works then we could compare the repos and see if there is something specific about that repo that may be a problem.  Thank you so much for your help in debugging this with us!

**Comment by KetchupOnMyKetchup** [comment]
I'm going to go ahead and close this issue since I don't see any new comments or pending issues and I've had the troubleshooting steps posted for a week since the bug fix went in. 

If anyone in the future runs into issues, please try these troubleshooting steps before opening a new issue. 
- https://github.com/github/vscode-github-actions/issues/77#issuecomment-1503960028 

Thank you all!

**Comment by spandan1460** [comment]
> @KetchupOnMyKetchup thanks for instructions, but it doesn't fix problem :(
> 
> I'm using `0.25.4` and `Code v1.77.3`. There is Github Actions output on the bottom:
> 
> ```
> Activating GitHub Actions extension...
> 
> Cannot fetch github context
> Error: Cannot fetch github context
> 	at Ao (/Users/i.mirdzhamolov/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/extension-node.js:2:1597413)
> 	at async /Users/i.mirdzhamolov/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/extension-node.js:2:1623054
> 	at async Rs (/Users/i.mirdzhamolov/.vscode/extensions/github.vscode-github-actions-0.25.4/dist/extension-node.js:2:1622125)
> 	at async E.n (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:96:6646)
> 	at async E.m (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:96:6609)
> 	at async E.l (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:96:6066)
> ```

Hello, I am facing the similar issues. I tried with all kinds of repos. Even I tried to login by creating a blank repo but still, I am getting the similar issue mentioned by @inomdzhon. 

I am using VSCode Version `v1.78` and github actions version `v0.25.7`.

Can anyone help me out on this?

**Comment by nicolasalt** [comment]
Same issue:

```
Cannot fetch github context
Error: Cannot fetch github context
	at Ao (/home/nz/.vscode-server/extensions/github.vscode-github-actions-0.25.7/dist/extension-node.js:2:1597788)
	at processTicksAndRejections (node:internal/process/task_queues:96:5)
	at async /home/nz/.vscode-server/extensions/github.vscode-github-actions-0.25.7/dist/extension-node.js:2:1634497
	at async Rs (/home/nz/.vscode-server/extensions/github.vscode-github-actions-0.25.7/dist/extension-node.js:2:1634453)
	at async v.n (/home/nz/.vscode-server/bin/b3e4e68a0bc097f0ae7907b217c1119af9e03435/out/vs/workbench/api/node/extensionHostProcess.js:97:6644)
	at async v.m (/home/nz/.vscode-server/bin/b3e4e68a0bc097f0ae7907b217c1119af9e03435/out/vs/workbench/api/node/extensionHostProcess.js:97:6607)
	at async v.l (/home/nz/.vscode-server/bin/b3e4e68a0bc097f0ae7907b217c1119af9e03435/out/vs/workbench/api/node/extensionHostProcess.js:97:6064)
```

**Comment by KetchupOnMyKetchup** [comment]
Is there something specific about the computer that it is being tried on (blocked by a network, other VS Code extensions affecting it)? Would it be possible to try it on another machine? 

Also just a head's up, sometimes there has been some caching issues with getting the new extension version with bug fixes and a restart has helped some folks.

**Comment by Harshgupta30** [comment]
i have the same issue You have not yet finished authorizing this extension to use GitHub. Would you like to try a different way? (device code)

**Comment by spandan1460** [comment]
@Harshgupta30, @nicolasalt Try reinstalling the VSCode and the extension, it worked for me interestingly. Hope this approach helps you guys 😉

**Comment by nicolasalt** [comment]
Re-installing didn't work, however then I went to Accounts and removed Github Actions access to my Github account. Then the extension re-requested access and started to work 🥳

**Comment by RD-1988** [comment]
same issue here

**Comment by barweiss45** [comment]
I ran into a similar issue with the same issue. I was getting:

```
Error getting GitHub context
Bad credentials
```
What worked for me was to click on manage trusted extensions, which can be found on the accounts menu (lower left-hand side), which brings up a Manage Trusted Extension toolbar at the top. 
![image](https://github.com/github/vscode-github-actions/assets/81769500/77fe1f12-7316-47b8-a3f0-00ce62c54c7f)
(sorry have my instance on the right, but by default, it would be on the left.

Next, I **unclicked** Github actions and then re-clicked it. The application made me go through the steps to re-authenticate and allow vscode access. 
![image](https://github.com/github/vscode-github-actions/assets/81769500/2c3d66f5-b6f3-4c3c-844f-53f82379e0a9)

After that, my issue was resolved. Hope that helps someone 😊

**Comment by benjaminmillhouse** [comment]
> I ran into a similar issue with the same issue. I was getting:
> 
> ```
> Error getting GitHub context
> Bad credentials
> ```
> 
> What worked for me was to click on manage trusted extensions, which can be found on the accounts menu (lower left-hand side), which brings up a Manage Trusted Extension toolbar at the top. ![image](https://user-images.githubusercontent.com/81769500/256937728-77fe1f12-7316-47b8-a3f0-00ce62c54c7f.png) (sorry have my instance on the right, but by default, it would be on the left.
> 
> Next, I **unclicked** Github actions and then re-clicked it. The application made me go through the steps to re-authenticate and allow vscode access. ![image](https://user-images.githubusercontent.com/81769500/256937883-2c3d66f5-b6f3-4c3c-844f-53f82379e0a9.png)
> 
> After that, my issue was resolved. Hope that helps someone 😊

This solution worked for me as well.  Thanks!

**Comment by sampathNG** [comment]
> Re-installing didn't work, however then I went to Accounts and removed Github Actions access to my Github account. Then the extension re-requested access and started to work 🥳

It worked bro 
👍

**Comment by jmsvarbas** [comment]
If someone has this problem, try putting this on terminal: code --open-external=https://github.com/login

**Comment by debanjanbasu** [comment]
> If someone has this problem, try putting this on terminal: code --open-external=https://github.com/login

This finally made my day. Behind a corp VPN.

**Comment by aksuharun** [comment]
Warning: 'open-external' is not in the list of known options, but still passed to Electron/Chromium.

**Comment by Messi002** [comment]
I have faced the same issues, it's mostly because you're behind a proxy(or VPN) either by you or your ISP(internet service provider), just try different ISPs if available

**Comment by shryay** [comment]
> I am able to login with v0.25.4 👍

Please help me.

**Comment by Sergio-bzh** [comment]
Do any one knows if this issue have been solved on Mac OS ?

I'm facing the same issue and even after removing and reinstalling VSCode I'm looping with the same kind of message :

"**_You have not yet finished authorizing this extension to use GitHub. Would you like to try a different way? (local server)_**"

Thank you in advance.

**Comment by tavlasogluahmetfurkan** [comment]
I was experiencing the same problem; I think I unknowingly changed my proxy settings. I fixed it with these steps:
1) Open Settings in VS Code by pressing Ctrl + , (comma).
2) Type "proxy" in the search box.
3) Completely empty the Http: Proxy section (don't write anything).
4) Uncheck the Http: Proxy Strict SSL box (Cancel).
5) Turn off the Http: Proxy Support option.
6) Close and reopen VS Code and try again. 


After that, complete all the requirements in the profile section at the bottom left.

That is, you've connected, but if there's a notification with 1 or 2 on top of the sign in the profile section at the bottom left, click on it and log in to every place you need to log in.
After fulfilling the initial requirements, you will only be logged into GitHub. If you log in to the other sections as well, you can use GitHub Copilot.

I hope I have been helpful. Have a good day.
